src/windows.rs

```diff
@@ -0,0 +1,39 @@
+extern crate winapi;
+use std::io::Error;
+use std::path::PathBuf;
+use std::mem;
+use self::winapi::um::winnt::HANDLE;
+use self::winapi::um::winbase::FILE_FLAG_BACKUP_SEMANTICS;
+use self::winapi::um::fileapi::BY_HANDLE_FILE_INFORMATION;
+
+/// uses winapi to get Windows file metadata
+pub fn windows_file_handle_info(pbuf: &PathBuf) -> Result<BY_HANDLE_FILE_INFORMATION, Error> {
+
+    extern "system" {
+        // https://msdn.microsoft.com/en-us/library/windows/desktop/aa363788(v=vs.85).aspx
+        fn GetFileInformationByHandle(a: HANDLE, b: *mut BY_HANDLE_FILE_INFORMATION) -> i32;
```

Comment:
You shouldn't need to declare this. winapi provides a declaration: https://docs.rs/winapi/0.3.4/x86_64-pc-windows-msvc/winapi/um/fileapi/fn.GetFileInformationByHandle.html